addons/sale/models/sale.py

```diff
@@ -34,6 +34,24 @@ def _amount_all(self):
                 'amount_total': amount_untaxed + amount_tax,
             })
 
+    @api.model
+    def _address_get(self):
```

Comment:
no need of this function. 
Directly you can put condition on qweb template.
something like : 

<t t-if="doc.partner_shipping_id == doc.partner_invoice_id and doc.partner_invoice_id != doc.partner_id">